src/app/settings/settings.reducer.spec.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  initialState,
+  settingsReducer,
+  ActionSettingsPersist,
+  SettingsActionTypes,
+  ActionSettingsChangeTheme,
+  ActionSettingsChangeAnimationsPage,
+  ActionSettingsChangeAnimationsPageDisabled,
+  ActionSettingsChangeAnimationsElements,
+  ActionSettingsChangeAutoNightMode
+} from './settings.reducer';
+
+describe('settingsReducer', () => {
+  it('ActionSettingsPersist with undefined new state should return the initial state', () => {
+    const action: ActionSettingsPersist = {
+      type: SettingsActionTypes.PERSIST,
+      payload: undefined
+    };
+    const state = settingsReducer(undefined, action);
+    expect(state).toBe(initialState);
+  });
+
+  it('ActionSettingsChangeTheme should return state with updated theme', () => {
+    const action = new ActionSettingsChangeTheme({ theme: 'dark' });
+    const state = settingsReducer(undefined, action);
+    expect(state.theme).toEqual('dark');
+  });
+
+  it('ActionSettingsChangeAnimationsPage should return state with updated pageAnimations', () => {
+    const action = new ActionSettingsChangeAnimationsPage({
+      pageAnimations: false
+    });
+    const state = settingsReducer(undefined, action);
+    expect(state.pageAnimations).toEqual(false);
+  });
+
+  it('ActionSettingsChangeAnimationsPageDisabled should return state with updated pageAnimationsDisabled', () => {
+    const action = new ActionSettingsChangeAnimationsPageDisabled({
+      pageAnimationsDisabled: true
+    });
+    const state = settingsReducer(undefined, action);
+    expect(state.pageAnimationsDisabled).toEqual(true);
```

Comment:
If you check implementation of the reducer this action actually does two things:

1. sets `pageAnimationsDisable` to `true`
2. sets `pageAnimations` to `false`

Both are important for the functionality. 

Background: page animations are disabled in IE and EDGE because they don't work, disabling is not enough though, they have to be turned off too. Disabling is there just to prevent user from re-enabling them in settings.